src/main/java/org/opensearch/knn/index/codec/KNNVectorsFormatParams.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.codec;
+
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import org.opensearch.knn.common.KNNConstants;
+
+import java.util.Map;
+
+/**
+ * Class provides params for LuceneHNSWVectorsFormat
+ */
+@NoArgsConstructor
+@Getter
+public class KNNVectorsFormatParams {
+    private int maxConnections;
+    private int beamWidth;
+
+    protected boolean validate(final Map<String, Object> params) {
```

Comment:
why this is protected? Shouldn't this be a public function?

Reply:
Now, after creating a separate package we need to make it public. Earlier protected is good enough